gap/semigroups/semieunit.gi

```diff
@@ -0,0 +1,719 @@
+    ErrorNoReturn("Semigroups: McAlisterTripleSemigroup: usage,\n",
+                  "every vertex of <X> must be in the orbit of some vertex ",
+                  "of <X> which is in <Y> - see the documentation ",
+                  "for more detail,");
+  fi;
+
+  for x in DigraphVertices(X) do
+    if not x in Union(orbs) and not (x in DigraphVertexLabels(Y)) then
+      ErrorNoReturn("Semigroups: McAlisterTripleSemigroup: usage,\n",
+                    "every vertex of <X> must be in the orbit of some ",
+                    "vertex of <X> which is in <Y> - see the documentation",
+                    " for more detail,");
+    fi;
+  od;
+
+  # Check condition M4 (essentially, check that G.Y = X is connected.)
+  min := DigraphVertexLabel(Y, DigraphSinks(DigraphRemoveLoops(Y))[1]);
+  if ForAny(GeneratorsOfGroup(Image(hom)), g -> min ^ g <> min) then
+      ErrorNoReturn("Semigroups: McAlisterTripleSemigroup: \n",
+                    "<act> must fix the vertex of <X> which is the minimal ",
+                    "vertex of <Y> - see the documentation for more detail,");
+  fi;
+
+  fam := NewFamily("McAlisterTripleSemigroupFamily",
+                   IsMcAlisterTripleSemigroupElement);
+
+  # Check if this McAlister triple semigroup is a monoid
+  if IsMeetSemilatticeDigraph(Y) then
+    filt := IsMcAlisterTripleSemigroup and IsMonoid;
+  else
+    filt := IsMcAlisterTripleSemigroup;
+  fi;
+
+  # Create the semigroup itself
+  M := Objectify(NewType(CollectionsFamily(fam), filt and IsAttributeStoringRep
+                         and IsEUnitaryInverseSemigroup and IsWholeFamily),
+                 rec());
+
+  M!.elementType := NewType(fam, IsMcAlisterTripleSemigroupElementRep);
+
+  SetMcAlisterTripleSemigroupGroup(M, G);
+  SetMcAlisterTripleSemigroupAction(M, anti_act);
+  SetMcAlisterTripleSemigroupPartialOrder(M, X);
+  SetMcAlisterTripleSemigroupSemilattice(M, Y);
+
+  GeneratorsOfSemigroup(M);
+  return M;
+end);
+
+InstallMethod(McAlisterTripleSemigroup,
+"for a perm group, digraph, and digraph",
+[IsPermGroup, IsDigraph, IsDigraph],
+function(G, X, Y)
+  return McAlisterTripleSemigroup(G, X, Y, OnPoints);
+end);
+
+InstallMethod(McAlisterTripleSemigroup,
+"for a perm group, digraph, homogeneous list, and action",
+[IsGroup, IsDigraph, IsHomogeneousList, IsFunction],
+function(G, X, sub_ver, act)
+  return McAlisterTripleSemigroup(G, X, InducedSubdigraph(X, sub_ver), act);
+end);
+
+InstallMethod(McAlisterTripleSemigroup,
+"for a perm group, digraph, and homogeneous list",
+[IsPermGroup, IsDigraph, IsHomogeneousList],
+function(G, X, sub_ver)
+  return McAlisterTripleSemigroup(G, X, InducedSubdigraph(X, sub_ver),
+                                  OnPoints);
+end);
+
+#############################################################################
+# Methods for McAlister triple semigroups
+#############################################################################
+# InstallMethod(\=, "for two McAlister triple semigroups",
```

Comment:
This commented out method should either be deleted, or you could include this in the isomorphism semigroup code if it would be worthwhile doing this.